src/repository/user-repository.ts

```diff
@@ -16,6 +16,10 @@ class UserRepository extends GenericRepository<User> {
 
         return user;
     }
+
+    async exists(email: string): Promise<User | null> {
+        return await this.findByEmail(email);
+    }
```

Comment:
como eu disse no comentário abaixo, por mim poderia continuar usando findByEmail, mas se for realmente usar esse exists, ele deveria retornar Promise<boolean> e não Promise<User | null>